worlds/shorthike/__init__.py

```diff
@@ -0,0 +1,169 @@
+        self.multiworld.itempool += [self.create_item("Silver Feather") for _ in range(self.options.silver_feathers - counter["Silver Feather"])]
+        self.multiworld.itempool += [self.create_item("Bucket") for _ in range(self.options.buckets - counter["Bucket"])]
+
+    def create_regions(self) -> None:
+        menu_region = Region("Menu", self.player, self.multiworld)
+        self.multiworld.regions.append(menu_region)
+        
+        main_region = Region("Hawk Peak", self.player, self.multiworld)
+
+        for loc in self.location_name_to_id.keys():
+            main_region.locations.append(ShortHikeLocation(self.player, loc, self.location_name_to_id[loc], main_region))
+
+        self.multiworld.regions.append(main_region)
+
+        menu_region.connect(main_region)
+
+        if self.options.goal == 0:
+            # Nap
+            self.multiworld.completion_condition[self.player] = lambda state: (state.has("Golden Feather", self.player, get_min_feathers(self, 7, 9))
+                or (state.has("Bucket", self.player) and state.has("Golden Feather", self.player, 7)))
+        elif self.options.goal == 1:
+            # Photo
+            self.multiworld.completion_condition[self.player] = lambda state: state.has("Golden Feather", self.player, 12)
+        elif self.options.goal == 2:
+            # Races
+            self.multiworld.completion_condition[self.player] = lambda state: (state.has("Golden Feather", self.player, get_min_feathers(self, 7, 9))
+                or (state.has("Bucket", self.player) and state.has("Golden Feather", self.player, 7)))
+        elif self.options.goal == 3:
+            # Help Everyone
+            self.multiworld.completion_condition[self.player] = lambda state: (state.has("Golden Feather", self.player, 12)
+                and state.has("Toy Shovel", self.player) and state.has("Camping Permit", self.player)
+                and state.has("Motorboat Key", self.player) and state.has("Headband", self.player)
+                and state.has("Wristwatch", self.player) and state.has("Seashell", self.player, 15)
+                and state.has("Shell Necklace", self.player))
+        elif self.options.goal == 4:
+            # Fishmonger
+            self.multiworld.completion_condition[self.player] = lambda state: (state.has("Golden Feather", self.player, get_min_feathers(self, 7, 9))
+                or (state.has("Bucket", self.player) and state.has("Golden Feather", self.player, 7))
+                and state.has("Fishing Rod", self.player))
+
+    def set_rules(self):
+        create_rules(self, location_table)
+
+    def fill_slot_data(self) -> Dict[str, Any]:
+        slot_data: Dict[str, Any] = {}
+        locations: Dict[int, Any] = {}
+
+        multiworld = self.multiworld
+        player = self.player
+        options = self.options
+
+        for loc in multiworld.get_filled_locations(player):
+            if loc.item.code == None:
```

Comment:
Just a minor nitpick, should be either `if not loc.item.code:` or `if loc.item.code is None:`